homeassistant/components/nissan_leaf/__init__.py

```diff
@@ -78,32 +78,60 @@
 SIGNAL_UPDATE_LEAF = 'nissan_leaf_update'
 
 SERVICE_UPDATE_LEAF = 'update'
+SERVICE_START_CHARGE_LEAF = 'start_charge'
 ATTR_VIN = 'vin'
 
 UPDATE_LEAF_SCHEMA = vol.Schema({
     vol.Required(ATTR_VIN): cv.string,
 })
+START_CHARGE_LEAF_SCHEMA = vol.Schema({
+    vol.Required(ATTR_VIN): cv.string,
+})
 
 
-async def async_setup(hass, config):
+def setup(hass, config):
     """Set up the Nissan Leaf component."""
     import pycarwings2
 
-    async def handle_update(service):
+    async def async_handle_update(service):
+        """Handle service to update leaf data from Nissan servers."""
         # It would be better if this was changed to use nickname, or
         # an entity name rather than a vin.
-        vin = service.data.get(ATTR_VIN, '')
+        vin = service.data[ATTR_VIN]
 
         if vin in hass.data[DATA_LEAF]:
             data_store = hass.data[DATA_LEAF][vin]
-            async_track_point_in_utc_time(
-                hass, data_store.async_update_data, utcnow())
-            return True
+            await data_store.async_update_data(utcnow())
+        else:
+            _LOGGER.debug("Vin %s not recognised for update", vin)
 
-        _LOGGER.debug("Vin %s not recognised for update", vin)
-        return False
+    async def async_handle_start_charge(service):
```

Comment:
Rather than a service, this seems like it may be a better fit for the switch platform.

Reply:
See previous conversation. It can't be turned off, so it's not a good fit for a switch.